### PR TITLE
Form handling improvements in import page step 1 and 2

### PR DESCRIPTION
**Pull request by rokaszygmantas**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Form handling improvements (not visible).
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | No QA needed yet, since the new import page is hidden and some additional improvements will come in the near future.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11160)

### CODE REVIEW

File: src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportFormDataProvider.php
Comment (rokaszygmantas) [inline src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportFormDataProvider.php:94]: @matks that's the only service in which session is injected and used now, but it is required in order to preselect the configuration values if user goes back from import step 2 to step 1. Is that reasonable? :)

**Comment by rokaszygmantas** [comment]
Hi @matks!

This is the first PR with import page improvements that we discussed on https://github.com/PrestaShop/PrestaShop/pull/10120.

I've refactored mostly the form handling part of both import step 1 and step 2.

There will be an additional PR at some point that implements CQRS for saving import data matches, now it's using legacy logic to save it (I mean this part: https://github.com/PrestaShop/PrestaShop/pull/11160/files#diff-f7f41ac7afb236175ee151c49f7359e6R66).

**Edit:** Actually I see at least two additional PRs coming after this:
1. CQRS implementation for data matches.
2. Import logic moved away from legacy controller into adapters.

File: src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportFormDataProvider.php
Comment (matks) [reply src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportFormDataProvider.php:94]: It is reasonable 😄 

However I'm curious: I see you introduced `ImportFormDataProviderInterface`. That is a good idea because this feature is so complex and specific. But since you created an Interface for this case, why not inject the session in ` getData()` and `setData()` ?

File: src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportFormDataProviderInterface.php
Comment (matks) [inline src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportFormDataProviderInterface.php:34]: It's good to have this interface 👍 this feature is so complex, it is relevant to have this specific interface for it

File: src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportFormHandler.php
Comment (matks) [inline src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportFormHandler.php:36]: We might merge this with a FormHandler for CRUD pages later, but it does not exist yet. Let's keep it in mind.

File: src/PrestaShopBundle/Resources/config/routing/admin/configure/advanced_parameters/import.yml
Comment (matks) [inline src/PrestaShopBundle/Resources/config/routing/admin/configure/advanced_parameters/import.yml:16]: 👍

**Review by matks — CHANGES_REQUESTED** [review]

**Comment by matks** [comment]
@rokaszygmantas Review done :) only minor issues

Agreed with the 2 incoming PRs: this is a good split of the workload 👍

File: src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportFormDataProvider.php
Comment (rokaszygmantas) [reply src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportFormDataProvider.php:94]: Well the interface does not limit a developer to use the session, it's only this particular implementation that requires the session. 
There already are two different import form data providers, one for step 1 (`ImportFormDataProvider`) and one for step 2 (`ImportDataConfigurationFormDataProvider`) and both implement the same `ImportFormDataProviderInterface`, while only one of them depends on the session.

**Comment by rokaszygmantas** [comment]
@matks thanks for review! I've left an answer to one of your comments, but other than that I didn't see any change requests, so waiting for your response or approval :slightly_smiling_face:

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
@rokaszygmantas you're right, sorry 😅 I did not see at the end of the review that there was actually no requested change

**Pull request merged**
